docs/how-to/hip_runtime_api/memory_management/coherence_control.rst

```diff
@@ -0,0 +1,151 @@
+.. meta::
+  :description: This chapter describes the coherence control of the HIP
+                ecosystem ROCm software.
+  :keywords: AMD, ROCm, HIP, host memory
+
+.. _coherence_control:
+
+*******************************************************************************
+Coherence control
+*******************************************************************************
+
+AMD GPUs host and device memory can have two different types of coherence:
+
+* **Coarse-grained coherence** means that memory is only considered up to date at 
+  kernel boundaries, which can be enforced through hipDeviceSynchronize,
+  hipStreamSynchronize, or any blocking operation that acts on the null
+  stream (e.g. hipMemcpy). For example, cacheable memory is a type of
+  coarse-grained memory where an up-to-date copy of the data can be stored
+  elsewhere (e.g. in an L2 cache).
+* **Fine-grained coherence** means the coherence is supported while a CPU/GPU 
+  kernel is running. This can be useful if both host and device are operating on
+  the same dataspace using system-scope atomic operations (e.g. updating an
+  error code or flag to a buffer). Fine-grained memory implies that up-to-date
+  data may be made visible to others regardless of kernel boundaries as
+  discussed above.
+
+.. note::
+
+  In order to achieve this fine-grained coherence, many AMD GPUs use a limited
+  cache policy, such as leaving these allocations uncached by the GPU, or making
+  them read-only.
+
+.. TODO: Is this still valid? What about Mi300?
+Developers should use coarse-grained coherence where they can to reduce
+host-device interconnect communication and also Mi200 accelerators hardware
+based floating point instructions are working on coarse grained memory regions.
```

Comment:
Where is this information from? I've never heard about floating-point (or any other instructions apart from memory accesses) requiring any form of coherency.

Reply:
Added to separate issue.